src/unistd.rs

```diff
@@ -1063,32 +1110,28 @@ pub fn gethostname() -> Result<OsString> {
 ///
 /// # Examples
 ///
-/// ```no_run
-/// use std::os::unix::io::AsRawFd;
-/// use nix::unistd::close;
-///
-/// let f = tempfile::tempfile().unwrap();
-/// close(f.as_raw_fd()).unwrap();   // Bad!  f will also close on drop!
-/// ```
-///
 /// ```rust
-/// use std::os::unix::io::IntoRawFd;
+/// use std::os::unix::io::AsFd;
 /// use nix::unistd::close;
 ///
 /// let f = tempfile::tempfile().unwrap();
-/// close(f.into_raw_fd()).unwrap(); // Good.  into_raw_fd consumes f
+/// close(f).unwrap();
 /// ```
-pub fn close(fd: RawFd) -> Result<()> {
-    let res = unsafe { libc::close(fd) };
+pub fn close<Fd: IntoRawFd>(fd: Fd) -> Result<()> {
```

Comment:
Nice.

Reply:
This may actually be wrong come to think of it. I think this lets you pass in an owned type which will double close. Need to double check. Solution is to ManuallyDrop before close.

Reply:
No, this is right.  `into_raw_fd` transfers ownership.
https://doc.rust-lang.org/stable/std/os/unix/io/trait.IntoRawFd.html

Reply:
Whew, forgot that `into_raw_fd` takes self by value. Then I agree, nice. :)